nautobot/dcim/models/device_components.py

```diff
@@ -648,9 +698,36 @@ def clean(self):
                 }
             )
 
-    @property
-    def parent(self):
-        return self.device
+        # Parent validation
+
+        # An interface cannot be its own parent
+        if self.pk and self.parent_interface_id == self.pk:
+            raise ValidationError({"parent_interface": "An interface cannot be its own parent."})
+
```

Comment:
Should this be grouped with the other `parent_interface` checks above, instead of sandwiched here between the `lag` and `bridge` checks?